Documentation/specs/runtime-configuration-file.md

```diff
@@ -123,6 +123,12 @@ This section is copied verbatim from an identical section in the input `project.
 
 These settings are read by `corehost` to determine how to initialize the runtime. All versions of `corehost` **must ignore** settings in this section that they do not understand (thus allowing new settings to be added in later versions).
 
+### `version` Property (`.deps.json`)
+
+This is a simple integer version indicating the version of the deps file format in use. It begins at `2` because the deps file format closely matches the format for `project.lock.json` which is at version `2`. In general, hosts should fail to load applications that use deps files with "unsupported" versions. It's up to the host to determine it's support policy. There is no guarantee of compatibility between versions of the deps file, each version could be completely different and require quirking to load properly, so no assumption should be made about the version field. The goal of this field is to allow us to make major changes without causing undefined behavior in existing hosts (instead providing a clearer "This application is not compatible with this host" error).
```

Comment:
The scenario I'm envisioning is publishing using tools that know about deps version 3, then taking that application and moving it to another machine that only has the older muxer (which only knows about deps version 2). It would be good if the user were to get a clear error message like "This application requires a newer version of the .NET Tools blah blah blah" rather than potentially undefined behavior.

Indeed, updating the deps file format mandates a change in the host, but how does the user know that the update is required? The idea here is to make that clear to the user. Similarly, it is likely that if a deps file change is made, we will want newer hosts to understand older formats for compatibility (or at least provide clear error messaging) and they might wish to change behavior based on this version field.